README.md

```diff
@@ -1,124 +1,182 @@
 
-## Other environment variables
+Given a YAML zonefile produce Terraform JSON for each specified provider. The produced Terraform is put in a directory called `tf-tmp/<provider>` 
 
-* BUCKET_NAME: S3 bucket name to store Terraform state file (default `dns-state-bucket-DEPLOY_ENV`)
+* ZONEFILE (required) - The YAML formatted file to use
+* PROVIDERS (required) - Which DNS providers to produce Terraform for
 
-## Generating Terraform DNS configuration
+### Providers ###
 
-We can currently generate DNS resource files for three providers:
+* `all` - Special value, uses all available providers.
+* `dyn` - [Dyn](https://dyn.com/)
+* `gce` - [Google's Cloud DNS](https://cloud.google.com/dns/docs/)
+* `route53` - [AWS' Route53](https://aws.amazon.com/route53)
 
-* Route53
-* GCE
-* Dyn - *NB* we will be using GCE in preference to Dyn
+## Terraform ##
 
-These files are generated in `tf-tmp/` and follow the naming scheme: `tf-tmp/<provider>/zone.tf`.
+A set of wrappers for standard Terraform commands. These wrappers store state in S3 using Terraform's remote state files.
 
-To generate the configuration files the following environment variables need to be set:
+* `tf:plan` - Calculate what changes would be made
+* `tf:validate` - Confirm that the terraform is valid (does not guarantee that it will work).
+* `tf:apply` - Make the changes
+* `tf:destroy` - Destroy the state.
 
-* ZONEFILE: the path to the YAML zone file
-* PROVIDERS: which providers to generate resources for, if more than one provider is wanted their names should be separated by commas or you can choose 'all'.
+### Shared environment variables ###
 
-To generate all resource files:
-```
-ZONEFILE=<path-to-zone-file> PROVIDERS=all bundle exec rake generate
-```
+Other than `tf:validate` the Terraform tasks all share a number of options (`tf:validate` which only needs `PROVIDERS`).
 
-To generate a specific resource file (for example Dyn's):
-```
-ZONEFILE=<path-to-zone-file> PROVIDERS=dyn bundle exec rake generate
-```
+* PROVIDERS (required) - see [above](#providers).
+* DEPLOY_ENV (required) - where to deploy to, one of `production`, `staging` or `integration`.
+* AWS_ACCESS_KEY_ID (required) - Access key with permissions for the bucket.
+* AWS_SECRET_ACCESS_KEY (required) - Secret associated with the access key.
+* AWS_DEFAULT_REGION - Which region the S3 bucket is in. Default: `eu-west-1`
+* BUCKET_NAME - Name of the S3 bucket. Default: `dns-state-bucket-<environment>`, `environment` is set by DEPLOY_ENV.
 
-## Environment variables for Terraform (explained)
+### Per provider environment variables ###
 
-* `DEPLOY_ENV` the environment to deploy to
-* `PROVIDERS` which providers to deploy (defaults to `all`)
+The specific providers each have specific variables that must be set
 
-#### Export the environment applicable to your credentials
-For example:
-```
-export DEPLOY_ENV=test
+#### Dyn ####
 
-export PROVIDERS=route53
-```
+* DYN_ZONE_ID - which zone to deploy to
```

Comment:
Can we have these in shell markdown ie `wrapped in backticks` to make it consistent with documentation above?